### DISCUSSION TITLE
[@types/node] Using `TestContext.assert` seems mis-typed versus `assert`

### DISCUSSION THREAD

**Discussion by fidian** [DefinitelyTyped/DefinitelyTyped#70477] [Issues with a @types package] [answered]
Node.JS 20 has support built-in for tests by default. [The documentation for `context.assert`](https://nodejs.org/api/test.html#contextassert) shows the example. [Playground Link](https://www.typescriptlang.org/play/?#code/JYWwDg9gTgLgBAbzjApgZ3gXzgMyhEOAcgDsIATFALlQyIG4AoR2mACiNaIBo42YAlHAC8APkSM4yAHRgANgEMSbAIwCmUmNIVo0KWNIxRgAYxgBRAI4BXBXP5RrKXjEcp1jTOqA)

```js
import { test } from 'node:test';

test('test', (t) => {
  t.plan(1);
  t.assert.strictEqual(true, true);
});
```

This produces the following error.

> Assertions require every name in the call target to be declared with an explicit type annotation.(2775)
>
> input.tsx(5, 15): 't' needs an explicit type annotation.

It would be great if all of these would work. [Playground Link](https://www.typescriptlang.org/play/?ssl=17&ssc=4&pln=1&pc=1#code/JYWwDg9gTgLgBAbzjApgZxgGjgFXTAYQgDtUAPeAXzgDMoIQ4ByYiAExQC5UMmBuAFChIsRHAgBrONToNmAQzRoUsAPQYowAMYwmAgatVwAQvQkpiAnjAAUTa02w2YASjgBeAHyIBcZADpFZVh-SWcoAFcUF0FKGP1DOAB1aAk0OAB3AAsLOBQyMAAbbWAYQoBPZHKwFDYrfDsHJxhOXHwiUnzXD28EXwCglRhQiXCo+LjBAyMUqDTMnOI4CLRgYgBzOGFoVDY4ACJJffqMRvxHOBs3Lx8-MJhI6NiYoA)

```js
import { test, TestContext } from 'node:test';
import { ok } from 'assert/strict'

// Broken
test('test', (t) => {
  t.assert.ok(true);
});

// Works when explicitly typed
test('test', (t: TestContext) => {
  t.assert.ok(true);
});

// Works when using imported "ok"
test('test', () => {
  ok(true);
});
```

The documentation for `context.assert` states the following:

> An object containing assertion methods bound to context. The top-level functions from the node:assert module are exposed here for the purpose of creating test plans.

@daviduzumeri [suggested that](https://github.com/microsoft/TypeScript/issues/59787#issuecomment-2330332206) this could be an issue with `@types/node` and how it's typing `TextContext`'s assert - `node:assert` is a full function but it's a lambda on `TestContext`. He also posted screenshots of the types.

[a comment by an automated account is not shown]

**Comment by Renegade334** [comment, marked as answer]
This is a TypeScript limitation relating to assertions. If you have an expression statement which calls a member function that returns an assertion, all identifiers in the property access chain leading to that function must be explicitly typed – this is mandated by the type checker, and there is no way around it.

In answer to your comment, the above applies regardless of whether the member type is declared as a method or as a property of function type.

Calling the top-level assert functions won't require the context parameter to be explicitly typed, but won't count the assertion against the test plan.
